sandbox/apparmor/apparmor.go

```diff
@@ -170,6 +170,35 @@ func ParserMtime() int64 {
 	return mtime
 }
 
+// PromptingSupported returns true if prompting is supported by the system.
+// Otherwise, returns false, along with a string explaining why prompting is
+// unsupported.
+func PromptingSupported() (bool, string) {
+	kernelFeatures, err := appArmorAssessment.KernelFeatures()
+	if err != nil {
+		return false, fmt.Sprintf("cannot check apparmor kernel features: %v", err)
+	}
+	parserFeatures, err := appArmorAssessment.ParserFeatures()
+	if err != nil {
+		return false, fmt.Sprintf("cannot check apparmor parser features: %v", err)
+	}
+	return PromptingSupportedByFeatures(kernelFeatures, parserFeatures)
+}
+
+// PromptingSupportedByFeatures returns whether prompting is supported by the
+// given AppArmor kernel and parser features.
+func PromptingSupportedByFeatures(kernelFeatures []string, parserFeatures []string) (bool, string) {
```

Comment:
I would slightly prefer this to take a struct, it's much clearer which is which in that case